### DISCUSSION TITLE
Rebase and merge in GH results in pipeline only applying single commit (in PR with multiple commits)

### DISCUSSION THREAD

**Discussion by umm0n** [gruntwork-io/knowledge-base#725] [Help]
Hello!

We're running the GW GHA pipeline in our ref-arch repo. We've noticed that when merging PRs with more than one commit, with multiple modified files, we can only use "squash and merge" or "create a merge commit" merge options. If we use "rebase and merge", only one of the commits is applied, leaving the rest unapplied. I've tested this behavior extensively. 

We try to avoid the default "create a merge commit" GH merge behavior, as this clutters our commit history with needless merge commits and difficults readability. We also often we create PR with multiple commits, affecting different directories. So, our current options are:

- Create a single commit per PR, and either "squash and merge" or "rebase and merge". This is not ideal, as it would significantly slow us down.
- Create a PR with multiple commits and "squash and merge". This is not ideal, considering commits in a same PR will often affects different directories/funciontality and having the same commit message for all of them is counter-intuitive.

Ideally, we would like to be able to "rebase and merge" PRs with multiple commits and have the pipeline apply each one of them. Is this feasible?

---

<ins datetime="2023-06-05T13:35:15Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110222">Tracked in ticket #110222</a></p>
</ins>

**Comment by denis256** [comment]
Hello,
executing pipeline for each commit may lead to destructive actions, in hard-to-spot cases of commits like:
* remove some resources 
* revert removal commit

Sequential execution of the pipeline will first remove resources from the removal commit and for next commit will try to create it back
